src/main/java/org/jasig/portlet/newsreader/adapter/RomeAdapter.java

```diff
@@ -198,22 +213,57 @@ public void init() throws Exception {
         // second field.
         compressingClient = new DecompressingHttpClient(httpClient);
     }
-
+    
+    protected String[] getPolicyPref(String... policyKey) {
+        String[] values = new String[policyKey.length];
+        
+        try {
+        
+            RequestAttributes requestAttributes = RequestContextHolder.currentRequestAttributes();
+            if (requestAttributes instanceof PortletRequestAttributes) {
+                PortletRequest request = ((PortletRequestAttributes) requestAttributes).getRequest();
+                PortletPreferences prefs = request.getPreferences();
+                List<String> valueList = new ArrayList<String>();
+                for (String key : policyKey) {
+                    valueList.add(prefs.getValue(key, defaultAntiSamyPolicy));
+                }
+                return valueList.toArray(values);
+            } 
+            
+            
+        } catch (IllegalStateException ex) {
+            
+            log.warn("Not Request Bound", ex);
+            
+        }
+        
+        
+        Arrays.fill(values, defaultAntiSamyPolicy);
```

Comment:
So if we're not inside a PortletRequest, we get the default policy applied across the board.